massgov/openmass
# Make conditional fields D10 compatible

**Description:**
See ticket. This also includes a JS fix on org edit pages.


**Jira:** (Skip unless you are MA staff)
DP-28040


**To Test:**
Org pages:
- Go to /orgs/qag-executive-office-of-technology-services-and-security/edit
- Click on content and go to the fourth organization section with a heading of "What would you like to do? 
- Select a display type of "Links" and leave the "Link group title" empty.
- Save
- Verify that there is an error
- Add a "Link group title" and it should save correctly.
- Repeat the process with a display type of "Buttons" and the title should not be required.

Promotional pages:
 - Go to /massachusetts-snap-restaurant-meals-program-rmp/edit
 - Click on content
 - Under the header section, switch between a solid color and single image background type.
 - Verify that background color is visible only when the background type is solid color.
 - Verify that the text box color is visible only when the background type is single image.

Iframe on location detail pages:
 - Go to /location-details/qag-locationdetails/edit
 - Click on sections
 - Scroll down to the iframe
 - Verify that "Set Fixed Height" shows only when height configuration is "Fixed height (default)"
 - Verify that "Set Aspect Ratio" shows only when height configuration is "Aspect ratio".
 - Verify that "Verify dynamic height configuration" shows only when height configuration is "Dynamic height".


---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)
